torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp

```diff
@@ -69,25 +69,124 @@ std::map<int64_t, Value*> InsertSymbolicShapesCompute(
-    bool add_composed_op);
+    bool add_composed_op,
+    std::vector<std::vector<StrideInput>>& input_info,
+    std::vector<StrideInput>& output_info);
+
+std::string toString(StrideInput si) {
+  switch (si) {
+    case StrideInput::TENSOR_CONT:
+      return "TENSOR_CONT";
+    case StrideInput::TENSOR_CONT_CHANNELS_LAST:
+      return "TENSOR_CONT_CHANNELS_LAST";
+    case StrideInput::S_ONE:
+      return "S_ONE";
+    case StrideInput::S_CONT:
+      return "S_CONT";
+    case StrideInput::S_TRAN_CONT:
+      return "S_TRAN_CONT";
+    case StrideInput::S_AS_ARG:
+      return "S_AS_ARG";
+  }
+  TORCH_INTERNAL_ASSERT(false);
+}
+
+StrideInput strideInputFromString(const std::string& si) {
+  if (si == "TENSOR_CONT") {
+    return StrideInput::TENSOR_CONT;
+  } else if (si == "TENSOR_CONT_CHANNELS_LAST") {
+    return StrideInput::TENSOR_CONT_CHANNELS_LAST;
+  } else if (si == "S_ONE") {
+    return StrideInput::S_ONE;
+  } else if (si == "S_CONT") {
+    return StrideInput::S_CONT;
+  } else if (si == "S_TRAN_CONT") {
+    return StrideInput::S_TRAN_CONT;
+  } else if (si == "S_AS_ARG") {
+    return StrideInput::S_AS_ARG;
+  } else {
+    TORCH_INTERNAL_ASSERT(false);
+  }
+}
+
+inline StrideInput summarizeStrideDim(
+    const c10::IntArrayRef sizes,
+    const c10::IntArrayRef strides,
+    size_t dim,
+    const std::vector<StrideInput>& stride_inputs,
+    size_t stride_inputs_offset) {
+  if (strides[dim] == 1) {
+    return StrideInput::S_ONE;
+  } else if (
+      dim + 1 < sizes.size() &&
+      strides[dim] == strides[dim + 1] * sizes[dim + 1]) {
+    return StrideInput::S_CONT;
+    // Transposed Contiguous depends on prior dim and contiguous depends on next
+    // dim, so to avoid a mutual dependence check that the next dim is Stride
+    // Contiguous
+  } else if (
+      dim > 0 && strides[dim] == strides[dim - 1] * sizes[dim - 1] &&
+      (stride_inputs[dim - 1 + stride_inputs_offset] != StrideInput::S_CONT)) {
+    return StrideInput::S_TRAN_CONT;
+  } else {
+    return StrideInput::S_AS_ARG;
+  }
+}
+
+std::vector<StrideInput> summarizeInputStrides(const TensorType& tt) {
+  auto strides = *tt.strides().concrete_sizes();
+  auto sizes = *tt.sizes().concrete_sizes();
+  if (c10::is_contiguous_strides(sizes, strides)) {
+    return {StrideInput::TENSOR_CONT};
+    // TODO: channels last 3d
+  } else if (c10::is_channels_last_strides_2d(sizes, strides)) {
+    return {StrideInput::TENSOR_CONT_CHANNELS_LAST};
+  }
+  std::vector<StrideInput> stride_inputs;
+  for (size_t dim = 0; dim < sizes.size(); ++dim) {
+    stride_inputs.push_back(
+        summarizeStrideDim(sizes, strides, dim, stride_inputs, 0));
+  }
+  return stride_inputs;
+};
+
+// Todo: incorporate in codegen
+StrideInput summarizeOutputStrides(const TensorType& tt) {
+  auto strides = *tt.strides().concrete_sizes();
+  auto sizes = *tt.sizes().concrete_sizes();
+  // We only try to maintain output striding for channels last tensors,
```

Comment:
What do you mean ?  In other cases, we will just make it contiguous

Reply:
we could add support for permuting to other ordering of dims but other than that would require more substantial changes

Reply:
We might have to support more generic striding because Hao mentioned that they come up in SR models.

Anyways, the point I had here was just to assert in other cases, so that we are doing something incorrectly. If we can't figure out the other cases here, then no problem.

Reply:
generic striding is already handle for inputs.. it's just that we don't preserve striding for outputs if they aren't contiguous or channels-last